### PR TITLE
Remove ApplyVertexModification in favor of ApplyMeshModification [Do not merge]

### PR DESCRIPTION
**Pull request by sebastienlagarde**
- Factor code between shader graph and inspectorUI, less duplication.
- Mean everything move to object space.

Note: This PR is required for GPU Terrain PR to simplify the code (https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/1687), it will allow to remove the undesired ApplyPreVertexModification and only have a single 'modification' call.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/HDRP/ShaderPass/VertMesh.hlsl [hunk 1]
```diff
@@ -96,7 +96,7 @@ VaryingsToDS InterpolateWithBaryCoordsToDS(VaryingsToDS input0, VaryingsToDS inp
 VaryingsMeshType VertMesh(AttributesMesh input)
 {
 #if defined(HAVE_MESH_MODIFICATION)
-    input = ApplyMeshModification(input);
+    input = ApplyMeshModification(input, _Time);
```
Comment (zeroyao) [inline com.unity.render-pipelines.high-definition/HDRP/ShaderPass/VertMesh.hlsl]: Terrain need instancing id to be set up before ApplyMeshModification. I moved this line after `UNITY_TRANSFER_INSTANCE_ID(input, output);`.

File: com.unity.render-pipelines.high-definition/HDRP/Editor/ShaderGraph/VertexAnimation.template.hlsl [hunk 2]
```diff
@@ -35,6 +35,12 @@ VertexDescriptionInputs AttributesMeshToVertexDescriptionInputs(AttributesMesh i
     return output;
 }
 
+AttributesMesh ApplyMeshModification(AttributesMesh input, float4 time)
```
Comment (cdxntchou) [inline com.unity.render-pipelines.high-definition/HDRP/Editor/ShaderGraph/VertexAnimation.template.hlsl]: Just merge these two functions, if the one below is not called anywhere else.

**Comment by sebastienlagarde** [comment]
Hey, some information. I failed the katana test with this PR, I have some issue with object scaling handling. It will take a little more work to get there, hope to have something Monday

**Comment by sebastienlagarde** [comment]
Close pull request. There is more work to do than this to be correct. Some change will be bring by Yao's PR instead.

**Pull request closed without merging**